### PR TITLE
added create skill endpoint

### PR DESCRIPTION
**Pull request by Glunky**

### CODE REVIEW

**Comment by Glunky** [comment]
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/848

**Review by BelehovEgor — COMMENTED** [review]
how i understand this task, you have to create new entity, another entity, i don't talk about DbSkill, for example DbUserSkill, which will be link entity. If Skill already exists you write id of found entity (DbSkill.Id) or you create new DbSkill and write its id

File: src/UserService.Data/SkillRepository.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            if (skill == null)
+            {
+                throw new ArgumentNullException(nameof(skill));
+            }
+
+            _provider.Skills.Add(skill);
+            _provider.Save();
+        }
+
+        public bool IsSkillAlreadyExist(string skillName)
```
[the remark this answers is not in the document]
Comment (Glunky) [reply src/UserService.Data/SkillRepository.cs]: ++

**Review by VLiamin — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public OperationResultResponse<Guid> Execute(CreateSkillRequest request)
+        {
+            OperationResultResponse<Guid> response = new();
+            Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+            DbUser dbUser = _userRepository.Get(senderId);
+
+            if (!(dbUser.IsAdmin || _accessValidator.HasRights(Rights.AddEditRemoveUsers)))
+            {
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:47]: add BadRequest http status code

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                response.Errors.Add("Not enough rights.");
+                response.Status = OperationResultStatusType.Failed;
+
+                return response;
+            }
+
+            if (_skillRepository.DoesSkillAlreadyExist(request.Name))
+            {
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:55]: add conflict http status code

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                response.Errors.Add($"Skill {request.Name} already exist");
+                response.Status = OperationResultStatusType.Conflict;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:57]: update kernel to 37 version and set status failed

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            DbUser dbUser = _userRepository.Get(senderId);
+
+            if (!(dbUser.IsAdmin || _accessValidator.HasRights(Rights.AddEditRemoveUsers)))
+            {
+                response.Errors.Add("Not enough rights.");
+                response.Status = OperationResultStatusType.Failed;
+
+                return response;
+            }
+
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:53]: add validation to skill

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            if (_skillRepository.DoesSkillAlreadyExist(request.Name))
+            {
+                response.Errors.Add($"Skill {request.Name} already exist");
+                response.Status = OperationResultStatusType.Conflict;
+
+                return response;
+            }
+
+            DbSkill skill = _mapper.Map(request);
+
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:63]: add http status "Created"

File: src/UserService.Data/Interfaces/ISkillRepository.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.UserService.Models.Db;
+
+namespace LT.DigitalOffice.UserService.Data.Interfaces
+{
+    [AutoInject]
+    public interface ISkillRepository
+    {
+        void Add(DbSkill skill);
```
Comment (MariyaSin) [inline src/UserService.Data/Interfaces/ISkillRepository.cs:9]: return skill id

File: src/UserService.Data/SkillRepository.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+{
+    public class SkillRepository : ISkillRepository
+    {
+        private readonly IDataProvider _provider;
+
+        public SkillRepository(IDataProvider provider)
+        {
+            _provider = provider;
+        }
+
```
Comment (MariyaSin) [inline src/UserService.Data/SkillRepository.cs:17]: remove empty line

File: src/UserService.Data/SkillRepository.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+        public void Add(DbSkill skill)
+        {
+            if (skill == null)
+            {
+                throw new ArgumentNullException(nameof(skill));
```
Comment (MariyaSin) [inline src/UserService.Data/SkillRepository.cs:23]: return null, do not throw exceptions

File: src/UserService.Data/SkillRepository.cs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+            }
+
+            _provider.Skills.Add(skill);
+            _provider.Save();
```
Comment (MariyaSin) [inline src/UserService.Data/SkillRepository.cs:27]: return skill.Id

File: src/UserService.Data/SkillRepository.cs [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        public bool DoesSkillAlreadyExist(string skillName)
+        {
+            if (string.IsNullOrEmpty(skillName))
+            {
+                throw new ArgumentException(nameof(skillName));
```
Comment (MariyaSin) [inline src/UserService.Data/SkillRepository.cs:34]: remove

File: src/UserService.Mappers/Db/DbSkillMapper.cs [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+        public DbSkillMapper(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public DbSkill Map(CreateSkillRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
```
Comment (MariyaSin) [inline src/UserService.Mappers/Db/DbSkillMapper.cs:23]: return null

File: src/UserService.Validation/Skill/CreateSkillRequestValidator.cs [hunk 13]
```diff
@@ -0,0 +1,16 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.UserService.Models.Dto;
+
+namespace LT.DigitalOffice.UserService.Validation.Skill
+{
+    public class CreateSkillRequestValidator : AbstractValidator<CreateSkillRequest>
+    {
+        public CreateSkillRequestValidator()
+        {
+            RuleFor(s => s.Name)
```
Comment (MariyaSin) [inline src/UserService.Validation/Skill/CreateSkillRequestValidator.cs:10]: add trim

File: src/UserService.Data/Interfaces/ISkillRepository.cs (hunk 7 above)
Comment (Glunky) [reply src/UserService.Data/Interfaces/ISkillRepository.cs:9]: why?) we know skill id before fnction was called

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 2 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:48]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 4 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs]: ++

File: src/UserService.Data/Interfaces/ISkillRepository.cs (hunk 7 above)
Comment (Glunky) [reply src/UserService.Data/Interfaces/ISkillRepository.cs]: ++

File: src/UserService.Data/SkillRepository.cs (hunk 8 above)
Comment (Glunky) [reply src/UserService.Data/SkillRepository.cs:17]: ++

File: src/UserService.Data/SkillRepository.cs (hunk 9 above)
Comment (Glunky) [reply src/UserService.Data/SkillRepository.cs]: ++

File: src/UserService.Data/SkillRepository.cs (hunk 10 above)
Comment (Glunky) [reply src/UserService.Data/SkillRepository.cs:26]: ++

File: src/UserService.Data/SkillRepository.cs (hunk 11 above)
Comment (Glunky) [reply src/UserService.Data/SkillRepository.cs]: ++

File: src/UserService.Mappers/Db/DbSkillMapper.cs (hunk 12 above)
Comment (Glunky) [reply src/UserService.Mappers/Db/DbSkillMapper.cs]: ++

File: src/UserService.Validation/Skill/CreateSkillRequestValidator.cs (hunk 13 above)
Comment (Glunky) [reply src/UserService.Validation/Skill/CreateSkillRequestValidator.cs]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+        {
+            OperationResultResponse<Guid> response = new();
+            Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+            DbUser dbUser = _userRepository.Get(senderId);
+
+            if (!(dbUser.IsAdmin || _accessValidator.HasRights(Rights.AddEditRemoveUsers)))
+            {
+                response.Errors.Add("Not enough rights.");
+                response.Status = OperationResultStatusType.Failed;
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:51]: status => firbiden

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+                response.Errors.Add($"Skill {request.Name} already exist");
+                response.Status = OperationResultStatusType.Failed;
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
+
+                return response;
+            }
+
+            DbSkill skill = _mapper.Map(request);
+
+            _skillRepository.Add(skill);
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:67]: your repo return Id, you can write response.Body = _skillRepository.Add(_mapper.Map); it is shortest

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+
+            _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+            response.Status = OperationResultStatusType.FullSuccess;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:70]: check if response.Body != null. Your repo can return null - it is not success

File: src/UserService.Mappers/Db/DbSkillMapper.cs [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            if (request == null)
+            {
+                return null;
+            }
+
+            return new DbSkill
+            {
+                Id = Guid.NewGuid(),
+                Name = request.Name,
```
Comment (MariyaSin) [inline src/UserService.Mappers/Db/DbSkillMapper.cs:29]: add trim

File: src/UserService/Controllers/SkillController.cs [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace LT.DigitalOffice.UserService.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class SkillController : ControllerBase
+    {
+        [HttpPost("create")]
+        public OperationResultResponse<Guid> Create
+            (
```
Comment (MariyaSin) [inline src/UserService/Controllers/SkillController.cs:15]: pull "(" to 15 string and reset tabs

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+    public class CreateSkillCommand : ICreateSkillCommand
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IUserRepository _userRepository;
+        private readonly IDbSkillMapper _mapper;
+        private readonly ISkillRepository _skillRepository;
+        private readonly IAccessValidator _accessValidator;
+
+        public CreateSkillCommand
+            (
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:26]: pull "(" to 25 string and reset tabs

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 14 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 15 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 16 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:72]: ++

File: src/UserService.Mappers/Db/DbSkillMapper.cs (hunk 17 above)
Comment (Glunky) [reply src/UserService.Mappers/Db/DbSkillMapper.cs]: ++

File: src/UserService/Controllers/SkillController.cs (hunk 18 above)
Comment (Glunky) [reply src/UserService/Controllers/SkillController.cs]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 19 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 5 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:59]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 20]
```diff
@@ -0,0 +1,92 @@
+﻿using LT.DigitalOffice.Kernel.AccessValidatorEngine.Interfaces;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:1]: use editor config

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
+
+                return response;
+            }
+
+            response.Body = _skillRepository.Add(_mapper.Map(request));
+
+            if(response.Body == default)
+            {
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:82]: i think it is not conflict

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+                response.Status = OperationResultStatusType.Failed;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:83]: you do not return, in this case in any way you get crated and fullSuccess status

File: src/UserService/Controllers/CommunicationController.cs [hunk 23]
```diff
@@ -29,14 +29,8 @@ public class CommunicationController : ControllerBase
         {
             OperationResultResponse<Guid> result = command.Execute(request);
```
Comment (MariyaSin) [inline src/UserService/Controllers/CommunicationController.cs:30]: return command.Execute(request). Add status response in command

File: src/UserService/Controllers/CredentialsController.cs [hunk 24]
```diff
@@ -26,19 +26,10 @@ public class CredentialsController : ControllerBase
             [FromServices] ICreateCredentialsCommand command,
             [FromBody] CreateCredentialsRequest request)
         {
-            var result = command.Execute(request);
+            OperationResultResponse<CredentialsResponse> result = command.Execute(request);
```
Comment (MariyaSin) [inline src/UserService/Controllers/CredentialsController.cs:29]: add status in command

File: src/UserService/Controllers/UsersController.cs [hunk 25]
```diff
@@ -30,19 +30,10 @@ public class UsersController : ControllerBase
...
-
-        return result;
-      }
-
-      if (result.Status != OperationResultStatusType.Failed)
-      {
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
-      }
+      _httpContextAccessor.HttpContext.Response.StatusCode = result.Status == OperationResultStatusType.Failed ?
+        (int)HttpStatusCode.Conflict : (int)HttpStatusCode.Created;
```
Comment (MariyaSin) [inline src/UserService/Controllers/UsersController.cs:36]: set status in command

Comment (Glunky) [reply src/UserService/Controllers/UsersController.cs]: ++

File: src/UserService/Controllers/CredentialsController.cs (hunk 24 above)
Comment (Glunky) [reply src/UserService/Controllers/CredentialsController.cs]: ++

File: src/UserService/Controllers/CommunicationController.cs (hunk 23 above)
Comment (Glunky) [reply src/UserService/Controllers/CommunicationController.cs]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 22 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:83]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 21 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:82]: ++

Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 20 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:1]: ++

File: src/UserService.Business/Commands/Communication/EditCommunicationCommand.cs [hunk 26]
```diff
@@ -66,9 +67,10 @@ public class EditCommunicationCommand : IEditCommunicationCommand
 
             if (valueOperation != null && _repository.IsCommunicationValueExist(valueOperation.value.ToString()))
             {
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/Communication/EditCommunicationCommand.cs:70]: add line

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs [hunk 27]
```diff
@@ -54,38 +60,42 @@ public OperationResultResponse<CredentialsResponse> Execute(CreateCredentialsReq
 
             OperationResultResponse<CredentialsResponse> response = new ();
 
-            var dbPendingUser = _userRepository.GetPendingUser(request.UserId);
+            DbPendingUser dbPendingUser = _userRepository.GetPendingUser(request.UserId);
             if (dbPendingUser == null)
             {
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs:66]: add line (all place in this file)

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs [hunk 28]
```diff
@@ -54,38 +60,42 @@ public OperationResultResponse<CredentialsResponse> Execute(CreateCredentialsReq
...
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
                 response.Status = OperationResultStatusType.Failed;
                 response.Errors.Add("Wrong password");
                 return response;
             }
 
             try
             {
-                var tokenResponse = _rcToken.GetResponse<IOperationResult<IGetTokenResponse>>(
+        IOperationResult<IGetTokenResponse> tokenResponse = _rcToken.GetResponse<IOperationResult<IGetTokenResponse>>(
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs:98]: what about tabulation

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 29]
```diff
@@ -286,21 +287,25 @@ private void SendEmail(DbUser dbUser, string password, List<string> errors)
     /// <inheritdoc/>
     public OperationResultResponse<Guid> Execute(CreateUserRequest request)
     {
+      OperationResultResponse<Guid> response = new();
+
       if (!(_accessValidator.IsAdmin() ||
         _accessValidator.HasRights(Rights.AddEditRemoveUsers)))
       {
-        throw new ForbiddenException("Not enough rights.");
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:295]: add line

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 30]
```diff
@@ -286,21 +287,25 @@ private void SendEmail(DbUser dbUser, string password, List<string> errors)
...
       _validator.ValidateAndThrowCustom(request);
 
-      OperationResultResponse<Guid> response = new();
-
       if (_userRepository.IsCommunicationValueExist(request.Communications.Select(x => x.Value).ToList()))
       {
-        response.Status = OperationResultStatusType.Conflict;
-        response.Errors.Add("Communication value already exist");
-        return response;
+          _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:305]: add line

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 31]
```diff
@@ -328,6 +333,7 @@ public OperationResultResponse<Guid> Execute(CreateUserRequest request)
 
       ChangeUserOffice(request.OfficeId, userId, response.Errors);
 
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:336]: line

**Review by BelehovEgor — APPROVED** [review]

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+      _validator = validator;
+    }
+
+    public OperationResultResponse<Guid> Execute(CreateSkillRequest request)
+    {
+      OperationResultResponse<Guid> response = new();
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser dbUser = _userRepository.Get(senderId);
+
+      if (!(dbUser.IsAdmin || _accessValidator.HasRights(Rights.AddEditRemoveUsers)))
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:51]: if notAdmin && if notHasRights => ....

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 33]
```diff
@@ -0,0 +1,94 @@
...
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+        response.Errors.AddRange(errors);
+        response.Status = OperationResultStatusType.Failed;
+
+        return response;
+      }
+
+      if (_skillRepository.DoesSkillAlreadyExist(request.Name))
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:69]: remove to validators

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs [hunk 34]
```diff
@@ -0,0 +1,94 @@
...
+      }
+
+      response.Body = _skillRepository.Add(_mapper.Map(request));
+
+      if (response.Body == default)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+        response.Status = OperationResultStatusType.Failed;
+
+        return response;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Skill/CreateSkillCommand.cs:85]: remove return and set 88 and 89 line before the "if"

File: src/UserService.Business/Commands/Communication/EditCommunicationCommand.cs (hunk 26 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Communication/EditCommunicationCommand.cs]: ++

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs (hunk 27 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs]: ++

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs (hunk 28 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs]: ++

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 29 above)
Comment (Glunky) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs:292]: ++

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 30 above)
Comment (Glunky) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs]: ++

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 31 above)
Comment (Glunky) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs:327]: ++

File: src/UserService.Business/Commands/Skill/CreateSkillCommand.cs (hunk 34 above)
Comment (Glunky) [reply src/UserService.Business/Commands/Skill/CreateSkillCommand.cs]: ++

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**